lib/wget.js

```diff
@@ -82,18 +83,28 @@ function download(src, output, options, _parentEvent, redirects) {
                 download(newLocation, output, options, downloader, redirects);
             }
 
-            if (res.statusCode === 200) {
+            if (res.statusCode === 200 || res.statusCode === 206) {
                 downloadedSize = 0;
                 fileSize = Number(res.headers['content-length']);
 
                 // If content length header is not sent, there is no way to determine file size.
                 if (isNaN(fileSize) === true) {
                     fileSize = null;
                 }
-                writeStream = fs.createWriteStream(output, {
-                    flags: 'w+',
+
+                const flags = res.statusCode === 206 ? 'a' : 'w+';
+                const writeStreamOptions = {
+                    flags,
                     encoding: 'binary'
-                });
+                };
+
+                if (!isNaN(options.start)) {
+                    fs.closeSync(fs.openSync(output, 'a'));
+                    writeStreamOptions.start = options.start;
+                    writeStreamOptions.flags = 'r+';
+                }
+
+                writeStream = fs.createWriteStream(output, writeStreamOptions);
```

Comment:
Should we have a try-catch for this?

Reply:
No need, as it bubbles up and should be handled by whichever is using this library.